lib/Star/Kata/Infrastructure/InMemoryInfrastructure.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * This file is part of the phpkata project.
+ *
+ * (c) Yannick Voyer (http://github.com/yvoyer)
+ */
+
+namespace Star\Kata\Infrastructure;
+
+use Star\Kata\KataDomain\KataService;
+use Star\Kata\Model\KataCollection;
+use Star\Kata\Model\KataRepository;
+
+/**
+ * Class InMemoryInfrastructure
+ *
+ * @author  Yannick Voyer (http://github.com/yvoyer)
+ *
+ * @package Star\Kata\Infrastructure
+ */
+final class InMemoryInfrastructure implements KataInfrastructure
+{
+    const CLASS_NAME = __CLASS__;
+
+    private $services = array();
+
+    /**
+     * @return KataService
+     */
+    public function kataService()
+    {
+        return $this->get(
+            'kata-service',
+            new KataService($this->kataRepository())
+        );
+    }
+
+    /**
+     * @return KataRepository
+     */
+    public function kataRepository()
+    {
+        return $this->get('kata-repository', new KataCollection());
+    }
+
+    private function get($id, $instance)
```

Comment:
This method is really strange.. it's called `get` which implies it's a query method, but then it acts as a command and mutates state by defining a service using the passed in `$instance`.. and then performs a query operation and returns the `$instance` you the caller already have back..?